spring-context/src/main/java/org/springframework/cache/support/CacheDecorator.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright 2002-2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cache.support;
+
+import org.springframework.cache.Cache;
+
+/**
+ * A functonal interface to allow us to decorate the {@code Cache} instances before they are added to the {@code org.springframework.cache.CacheManager}.
+ *
+ * @author Petar Tahchiev
+ * @since 5.1.0
+ */
+public interface CacheDecorator {
```

Comment:
Should be flagged with `@FunctionalInterface`. Not sure I would add that in the description of the interface.